clients/cobol-lsp-vscode-extension/src/services/reporter/index.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2025 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import { TelemetryEvent, TelemetryEventMeasurements } from "./model";
+import TelemetryReporter from "@vscode/extension-telemetry";
+
+let reporter: TelemetryReporter | undefined;
+
+/**
+ * for tests
+ */
+export function setReporter(r: TelemetryReporter | undefined): void {
+  reporter = r;
+}
+
+export async function initTelemetry(context: vscode.ExtensionContext) {
+  const telemetryKey = await getTelemetryKey(context);
+  if (telemetryKey === undefined) {
+    return;
+  }
+  reporter = new TelemetryReporter(telemetryKey);
+  context.subscriptions.push(reporter);
+}
+/**
+ * This method collect the data event for telemetry purpose, apply basic validation
+ * and delegate the reporter service to sent them to the target telemetry server
+ * @param eventName name assigned for the event for the telemetry
+ * @param categories optional list of keywords (act as tags)
+ * @param notes optional brief description
+ * @param telemetryMeasurement optional set of numeric data with a key name
+ */
+export function registerEvent(
+  eventName: string,
+  categories?: string[],
+  notes?: string,
```

Comment:
either fix the signature here and on registerExceptionEvent or on createTelemetryEvent.

Reply:
The `categories` parameter still need fixing.